app/models/concerns/foreman_teamdynamix/host_managed_extensions.rb

```diff
@@ -0,0 +1,21 @@
+module ForemanTeamdynamix
+  module HostManagedExtensions
+    extend ActiveSupport::Concern
+
+    def td_api
+      @td_api || TeamdynamixApi.new
+    end
+
+    included do
+      after_create :create_teamdynamix_asset
+    end
+
+    def create_teamdynamix_asset
+      asset = td_api.create_asset(self)
```

Comment:
The td_api should throw exceptions if it fails to create the asset in TD. This will cause the callback to fail and rollback the db transaction causing the host record to not be saved. This is what we want. Can you verify that this does indeed cause the host to not be saved, if the api fails to create the asset in TD?

Reply:
See "6 Halting Execution" here: http://guides.rubyonrails.org/active_record_callbacks.html#6

Reply:
I have tested this, it works as expected. TD Api raises an exception on any type of failure during create call. This JSONize exception (containing HTTP status and the error message) is automatically handled as an exception during callback transaction chain, as explained in rails guide. Host is not saved, and an error trace is displayed as per foreman code design.